tests/test_api.py

```diff
@@ -38,132 +41,171 @@ def test_phone_registration_sends_message(client, mocker):
+        assert response.status_code == 400
+        response_data = json.loads(json.dumps(response.data))
+        assert response_data["session_token"][0] == "This field is required."
+        assert response_data["security_code"][0] == "This field is required."
+
+        data = {"security_code": SECURITY_CODE}
+        response = client.json.post(url, data=data)
+        assert response.status_code == 400
+        response_data = json.loads(json.dumps(response.data))
+        assert response_data["phone_number"][0] == "This field is required."
+
+
+def test_phone_verification_with_incorrect_payload(client, backend):
+    with override_settings(PHONE_VERIFICATION=backend):
+        f.create_verification(
+            security_code=SECURITY_CODE,
+            phone_number=PHONE_NUMBER,
+            session_token=SESSION_TOKEN,
+        )
+        url = reverse("phone-verify")
+        # Payload with wrong session token
+        data = {
+            "phone_number": PHONE_NUMBER,
+            "security_code": SECURITY_CODE,
+            "session_token": "wrong-session-token",
+        }
+        response = client.json.post(url, data=data)
+        response_data = json.loads(json.dumps(response.data))
+        if backend.get("BACKEND").split("phone_verify.backends.")[1] in backends:
+            assert response.status_code == 400
+            assert response_data["non_field_errors"][0] == "Session Token mis-match"
+        elif (
+            backend.get("BACKEND").split("phone_verify.backends.")[1]
+            in sandbox_backends
+        ):
+            # Sandbox Backend returns a 200 status code as it does not check the payload information
+            assert response.status_code == 200
+
+        # Payload with wrong security code
+        data = {
+            "phone_number": PHONE_NUMBER,
+            "security_code": "999999",
+            "session_token": SESSION_TOKEN,
+        }
+        response = client.json.post(url, data=data)
+        response_data = json.loads(json.dumps(response.data))
+        if backend.get("BACKEND").split("phone_verify.backends.")[1] in backends:
+            assert response.status_code == 400
+            assert response_data["non_field_errors"][0] == "Security code is not valid"
+        elif (
+            backend.get("BACKEND").split("phone_verify.backends.")[1]
+            in sandbox_backends
+        ):
+            # Sandbox Backend returns a 200 status code as it does not check the payload information
+            assert response.status_code == 200
+
+        # Payload with incorrect phone_number
+        data = {
+            "phone_number": "+13478379632",
+            "security_code": SECURITY_CODE,
+            "session_token": SESSION_TOKEN,
+        }
+        response = client.json.post(url, data=data)
+        response_data = json.loads(json.dumps(response.data))
+        if backend.get("BACKEND").split("phone_verify.backends.")[1] in backends:
```

Comment:
How do the above statements, affect this statement? This is just used to check if the `BACKEND` setting is not a SandboxBackend. Because things work differently when working with normal Backend and SandboxBackend(specifically the `validate_security_code` method, as it does not perform any validation. )

Reply:
You pick backends from conftest file and then just normally compare backend value with actual value. No need for split?

Reply:
I check for the availability of backend in `backends` list which is `["twilio.TwilioBackend", "nexmo.NexmoBackend"]`. However, the `BACKEND` defined in settings would be like `phone_verify.backends.twilio.TwilioBackend`. So, I split it so that I get the part `twilio.TwilioBackend` and then compare if it exists in the list or not.

Another method was to explicitly compare the backends using `or` condition with the whole string which would become clumsy if more backends are added. So I came up with this approach.

Reply:
How about providing the class instead of a string?

Reply:
Do you mean in comparison? I am not sure exactly how are you thinking to implement it. The settings will always hold the `BACKEND` as a string, so it has to be compared with a string. No?

Reply:
Commented above.

Reply:
https://github.com/CuriousLearner/django-phone-verify/pull/40#discussion_r384689434